components/tag/index.css

```diff
@@ -265,26 +251,32 @@
 		padding-block-start: calc(var(--mod-tag-label-spacing-block, var(--spectrum-tag-label-spacing-block)) - var(--mod-tag-border-width, var(--spectrum-tag-border-width)));
 
 		/* space from label text to end edge */
-		margin-inline-end: calc(var(--mod-tag-label-spacing-inline-end, var(--spectrum-tag-label-spacing-inline-end)) - var(--mod-tag-border-width, var(--spectrum-tag-border-width)));
+		margin-inline-end: calc(var(--mod-tag-label-spacing-inline, var(--spectrum-tag-label-spacing-inline)) - var(--mod-tag-border-width, var(--spectrum-tag-border-width)));
 	}
 
-	&:hover {
-		border-color: var(--highcontrast-tag-border-color-hover, var(--mod-tag-border-color-hover, var(--spectrum-tag-border-color-hover)));
-		background-color: var(--highcontrast-tag-background-color-hover, var(--mod-tag-background-color-hover, var(--spectrum-tag-background-color-hover)));
-		color: var(--highcontrast-tag-content-color-hover, var(--mod-tag-content-color-hover, var(--spectrum-tag-content-color-hover)));
+	&:not(.is-disabled, .is-readOnly):hover {
+		/* mod/highcontrast specified at top */
+		border-color: var(--spectrum-tag-border-color-hover);
+		background-color: var(--spectrum-tag-background-color-hover);
+		color: var(--spectrum-tag-content-color-hover);
 	}
 
-	&:active {
-		border-color: var(--highcontrast-tag-border-color-active, var(--mod-tag-border-color-active, var(--spectrum-tag-border-color-active)));
-		background-color: var(--highcontrast-tag-background-color-active, var(--mod-tag-background-color-active, var(--spectrum-tag-background-color-active)));
-		color: var(--highcontrast-tag-content-color-active, var(--mod-tag-content-color-active, var(--spectrum-tag-content-color-active)));
+	&:not(.is-disabled, .is-readOnly):active {
+		/* mod/highcontrast specified at top */
+		border-color: var(--spectrum-tag-border-color-active);
+		background-color: var(--spectrum-tag-background-color-active);
+		color: var(--spectrum-tag-content-color-active);
+
+		/* stylelint-disable-next-line spectrum-tools/no-unknown-custom-properties -- height and width are set by implementations */
+		transform: perspective(max(var(--spectrum-downstate-height), var(--spectrum-downstate-width) * var(--spectrum-component-size-width-ratio-down))) translateZ(var(--spectrum-component-size-difference-down));
 	}
 
-	&:focus-visible,
-	&.is-focused {
-		border-color: var(--highcontrast-tag-border-color-focus, var(--mod-tag-border-color-focus, var(--spectrum-tag-border-color-focus)));
-		background-color: var(--highcontrast-tag-background-color-focus, var(--mod-tag-background-color-focus, var(--spectrum-tag-background-color-focus)));
-		color: var(--highcontrast-tag-content-color-focus, var(--mod-tag-content-color-focus, var(--spectrum-tag-content-color-focus)));
+	&:not(.is-disabled, .is-readOnly):focus-visible,
+	&:not(.is-disabled, .is-readOnly).is-focused {
```

Comment:
Do we still need `.is-focused` for S2 if it's already handled by the `:focus-visible`? Is this a legacy thing? If needed for the implementation to apply, it should probably be documented.